git-secrets

```diff
@@ -106,6 +110,48 @@ scan_history() {
   process_output $? "${output}"
 }
 
+# Takes an argument like commitHashA..commitHashB
+# Scans modified files in each commit at each point in history
+scan_between() {
+  # Fail if unstaged changes exist
+  git update-index --refresh
+  [[ -n $(git status --porcelain=v1 2> /dev/null) ]] &&
+    printf "Commit or stash changes first." >&2 && exit 1
```

Comment:
If the checked out repo is modified during CI in any way, this will fail. Potential problem.

Reply:
I have solved this, stand by shortly as I will push the change.